cmd/kubeadm/app/componentconfigs/kubelet_unix_test.go

```diff
@@ -0,0 +1,65 @@
+//go:build !windows
+// +build !windows
+
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package componentconfigs
+
+import (
+	"reflect"
+	"testing"
+
+	kubeletconfig "k8s.io/kubelet/config/v1beta1"
+	"k8s.io/utils/ptr"
+)
+
+func TestMutateResolverConfig(t *testing.T) {
+	var resolverConfig *string
+	if isSystemdResolvedActive, _ := isServiceActive("systemd-resolved"); isSystemdResolvedActive {
```

Comment:
something to improve here is to mock isServiceActive during unit tests. ideally UT should not depend on the unerlying system setup.

```
mutateResolverConfig(cfg *kubeletconfig.KubeletConfiguration, isServiceActiveFunc (string) (bool, error)) error {
```

then during actual runs isServiceActiveFunc can be passed  isServiceActive and during UT it can be a test case stub that returns true or false.

so there could be 2 test cases.

Reply:
Good idea, thank